lms/services/digest/_models.py

```diff
@@ -0,0 +1,76 @@
+from dataclasses import dataclass
+from functools import cached_property
+from typing import List
+
+
+@dataclass
+class HUser:
+    """A class representing H users."""
+
+    h_userid: str
+    """The H userid like 'acct:name@lms.example.com'."""
+
+    display_name: str
+    """A human readable display name."""
+
+    email: str
+    """Email for the user."""
+
+
+@dataclass
+class HCourse:
+    """A class representing courses as groups in H."""
+
+    title: str
+    """Title for the course."""
+
+    authority_provided_id: str
+    """The course's authority provided id."""
+
+    aka: List[str]
+    """All authority provided ids which map to the course from groups etc."""
+
+    instructors: List[str]
+    """A list of H userids for users who are instructors for the course."""
+
+    annotations: List[dict] = None
+    """A list of annotations in the course."""
+
+    @cached_property
+    def learner_annotations(self):
+        """Get the annotations, but filtered to users who are learners."""
+
+        return [
+            annotation
+            for annotation in self.annotations
+            if annotation["author"]["userid"] not in self.instructors
+        ]
```

Comment:
Previously this was part of the code which generated the courses in the first place, but it doesn't really seem like it's anyone's business but our own.

This keeps the outside code simpler, but it also means that should we want to know the total number of annotations as well as the number from students, we can. In general I think it's more flexible to store all the info, and then whittle it down for display.